lib/shopify_api/rest/base.rb

```diff
@@ -234,16 +234,20 @@ def create_instance(data:, session:, instance: nil)
             attr_sym = attribute.to_sym
 
             if has_many?(attr_sym) && value
+              instance.original_state[attr_sym] = []
               attr_list = []
               value.each do |element|
-                attr_list << T.unsafe(@has_many[attr_sym]).create_instance(data: element, session: session)
+                child = T.unsafe(@has_many[attr_sym]).create_instance(data: element, session: session)
+                attr_list << child
+                instance.original_state[attr_sym] << child.to_hash(true)
               end
               instance.public_send("#{attribute}=", attr_list)
             elsif has_one?(attr_sym) && value
               # force a hash if core returns values that instantiate objects like "USD"
               data_hash = value.is_a?(Hash) ? value : { attribute.to_s => value }
-              instance.public_send("#{attribute}=",
-                T.unsafe(@has_one[attr_sym]).create_instance(data: data_hash, session: session))
+              child = T.unsafe(@has_one[attr_sym]).create_instance(data: data_hash, session: session)
+              instance.public_send("#{attribute}=", child)
+              instance.original_state[attr_sym] = child.to_hash(true)
```

Comment:
This was a breaking change for our application that went unreported in the changelog. We were creating an instance and immediately calling save on it, which with this change resulted in an empty array being sent to Shopify.

Reply:
Not sure if we were using the library improperly since I didn't see mention of this is the docs, but if not I can go update the changelog for others who might run into this.